src/status_im2/contexts/syncing/enter_sync_code/view.cljs

```diff
@@ -0,0 +1,79 @@
+(ns status-im2.contexts.syncing.enter-sync-code.view
+  (:require [clojure.string :as string]
+            [quo2.core :as quo]
+            [quo2.foundations.colors :as colors]
+            [reagent.core :as reagent]
+            [react-native.core :as rn]
+            [react-native.clipboard :as clipboard]
+            [utils.i18n :as i18n]
+            [status-im2.contexts.syncing.enter-sync-code.style :as style]
+            [utils.debounce :as debounce]
+            [utils.re-frame :as rf]
+            [status-im2.contexts.syncing.utils :as sync-utils]))
+
+(defn view
+  []
+  (let [default-value           (reagent/atom nil)
+        submit-button-disabled? (reagent/atom true)
+        profile-color           (rf/sub [:profile/customization-color])]
+    (fn []
+      (let [invalid?           false
+            show-paste-button? (string/blank? @default-value)]
+        [:<>
+         [rn/view
+          {:style style/label-container}
+          [quo/text
+           {:style  style/label-pairing
+            :weight :medium
+            :size   :paragraph-2}
+           (i18n/label :t/type-pairing-code)]]
+         [rn/view {:style style/container-text-input}
+          [rn/view {:style (style/text-input-container invalid?)}
+           [rn/text-input
+            {:style                  (style/text-input)
+             :default-value          @default-value
+             :placeholder            (i18n/label :t/type-some-chat-key)
+             :on-change-text         (fn [scan-code]
+                                       (reset! default-value scan-code)
+                                       (reset! submit-button-disabled? false))
+             :blur-on-submit         true
+             :return-key-type        :done
+             :accessibility-label    :enter-sync-code-input
+             :auto-capitalize        :none
+             :placeholder-text-color colors/white-opa-40
+             :multiline              true}]
+           (if show-paste-button?
+             [rn/view {:style style/button-paste}
+              [quo/button
+               {:on-press (fn [_]
+                            (clipboard/get-string #(reset! default-value %))
+                            (reset! submit-button-disabled? false))
```

Comment:
wondering why we want to enable submit button here, what if the value is empty? or invalid?

Reply:
Hmm I did not test case when clipboard is empty 😄 
I'll take a look

Reply:
Thanks for the suggestion, Improved it here : https://github.com/status-im/status-mobile/pull/16852/commits/8f9c098f63d5f0e42a3ed432febb637b2dd20408